code/modules/client/client procs.dm

```diff
@@ -140,6 +140,11 @@
 		admins += src
 		holder.owner = src
 
+	var/localhost_addresses = list("127.0.0.1","::1")
+	if(!address || (address in localhost_addresses))
```

Comment:
While doing the !address bit is required to give people who launch the code in DS standalone host, its the biggest thing you might want to worry about.

There was already once an exploit that allowed somebody make their address and cid in isbanned() be null, there might be another that effects client.version, allowing somebody to spoof.

Change this to `if ((!address && !world.port) || ...` to be more secure.